scalingo/data_scalingo_container_size.go

```diff
@@ -0,0 +1,103 @@
+package scalingo
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/Scalingo/go-scalingo/v5"
+)
+
+func dataSourceScContainerSize() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceScContainerSizeRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"sku": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"human_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"human_cpu": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"memory": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"pids_limit": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"hourly_price": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"thirtydays_price": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"swap": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"ordinal": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+// dataSourceScNotificationPlatformRead performs the Scalingo API lookup
+func dataSourceScContainerSizeRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, _ := meta.(*scalingo.Client)
+
+	containers, err := client.ContainerSizesList(ctx)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	name, _ := d.Get("name").(string)
+
+	i := 0
+	for i < len(containers) && containers[i].Name != name {
+		i++
+	}
+	if i >= len(containers) {
+		return diag.Errorf("container '%v' not found", name)
+	}
```

Comment:
We usually never use  `for` loops in Go 😅 👴🏾  I would replace it with a for range loop

Reply:
we use `for` loops, but not with a `i := 0` and i++`